### PR TITLE
File parsing

### PR DESCRIPTION
**Pull request by j-asefa**
Working ephemeris file parser. There is some error checking still to do and a few test cases to add, but you should be able to work off of this branch and just pass a properly formatted ephemeris file to the `parse_ephemeris_file()` function.

Also, I was having some trouble squashing the commits for this PR, but will try to keep things clean in the future.

### CODE REVIEW

File: kaos/parser.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+Parsed files are stored in the DB.
+"""
+import os
+from collections import namedtuple
+from kaos.models import DB, OrbitRecords, SatelliteInfo, OrbitSegments
+
+OrbitPoint = namedtuple('OrbitPoint', 'time, pos, vel')
+
+def add_segment_to_db(orbit_data, satellite_id):
+    """Add the given segment to the database.
```
Comment (ZeyadTamimi) [inline kaos/parser.py]: Eh are you aligning to 100 chars, apply comment to entire file :P

File: kaos/parser.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        # TODO Validate uniqueness for this platform
+        orbit_record = OrbitRecords(platform_id=satellite_id, segment_id=segment.segment_id,
+                                    time=orbit_point.time, position=orbit_point.pos,
+                                    velocity=orbit_point.vel)
+        orbit_record.save()
+
+    DB.session.commit()
+
+def parse_ephemeris_file(file_handle):
+    """Parse the given ephemeris file and store
```
Comment (ZeyadTamimi) [inline kaos/parser.py]: Are you aligning to 100 chars?

File: kaos/parser.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    for orbit_point in orbit_data:
+        # TODO Validate uniqueness for this platform
+        orbit_record = OrbitRecords(platform_id=satellite_id, segment_id=segment.segment_id,
+                                    time=orbit_point.time, position=orbit_point.pos,
+                                    velocity=orbit_point.vel)
+        orbit_record.save()
+
+    DB.session.commit()
+
+def parse_ephemeris_file(file_handle):
```
Comment (ZeyadTamimi) [inline kaos/parser.py]: Not really a file handle, its literally a path

File: kaos/parser.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    the orbital data into the DB"""
+
+    #TODO: pre-process the file. Verify file formatting is as we expect
+
+    sat = SatelliteInfo(platform_name=os.path.splitext(file_handle)[0])
+    sat.save()
+    DB.session.commit()
+
+    with open(file_handle, "rU") as f:
+        """A list containing each of the 15 or so
```
Comment (ZeyadTamimi) [inline kaos/parser.py]: Comments as to what the variable is should not be there. The name should make it obvious ( which it does )

File: kaos/parser.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+                read_segment_boundaries = True
+
+            if "END Ephemeris" in line:
+                add_segment_to_db(segment_tuples, sat.platform_id)
+                read_orbital_data = False
+
+            if read_orbital_data:
+                line = line.strip()
+                if line:
+                    """each row is a 7-tuple formatted as
```
Comment (ZeyadTamimi) [inline kaos/parser.py]: Should probably be int he doc of the function since its basically  precondition on the file. Unsure

File: kaos/parser.py [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+                    """ The line we just read is a segment boundary,
+                    So first check that this is the *end* of a segment and then
+                    add this segment to the db.
+
+                    A segment begins with the same time-stamp that the previous segment
+                    ended with. So we don't want to commit the first entry in a
+                    *new* segment. The last_seen_segment_boundary != orbit_tuple.time checks
+                    this and makes sure not to commit to the db in this case."""
+                    if orbit_tuple.time in segment_boundaries:
```
Comment (ZeyadTamimi) [inline kaos/parser.py]: And these two conditions

File: kaos/parser.py [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+                        if last_seen_segment_boundary != orbit_tuple.time:
+                            last_seen_segment_boundary = orbit_tuple.time
+                            add_segment_to_db(segment_tuples, sat.platform_id)
+                            del segment_tuples[:]
```
Comment (ZeyadTamimi) [inline kaos/parser.py]: Not sure about using del, just set to empty segment_tuples = []

File: test/test_parser.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+from . import KaosTestCase
+from .context import kaos
+from kaos.models import *
+from kaos.parser import *
+
+class TestEphemerisParser(KaosTestCase):
+    """Ensures that the response history table behaves as expected."""
+
+    def test_simple_parser_J2000(self):
+        parse_ephemeris_file("ephemeris/Radarsat2_J2000.e")
```
Comment (ZeyadTamimi) [inline test/test_parser.py]: Test number of segments, and the min max time!

File: test/test_parser.py [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+        orbit = OrbitRecords()
+
+        # test that the correct number of entries was created
+        self.assertTrue(len(orbit.query.all()) == 17307) #taken from ephem file
+
+    def test_simple_parser_Fixed(self):
+        parse_ephemeris_file("ephemeris/Radarsat2_Fixed.e")
```
Comment (ZeyadTamimi) [inline test/test_parser.py]: Ditto

**Comment by j-asefa** [comment]
@ZeyadTamimi I think I addressed all of your concerns. There is one test case that is broken. It's under `test_ephemeris_parser_single_file` in `test_parser.py`. 

This test fails at `self.assertTrue(q.start_time == seg_start)`, and I'm not sure why. I've printed both values extensively and they seem to be the exact same. I'm at a loss as to why it's failing.

File: test/test_parser.py [hunk 10]
```diff
@@ -1,26 +1,147 @@
...
+			       6.8123706566399999e+05, 7.3718320319999999e+05,
+			       8.5152938083199994e+05, 9.3618207907199999e+05,
+			       1.0368000000000000e+06 ]
+
+	seg_times = [float(seg) for seg in segment_boundaries]
+
+	query = orbit_segment.query.all()
+
+	for q, seg_start in zip(query, seg_times[:-1]):
+	    self.assertTrue(q.start_time == seg_start)
```
Comment (j-asefa) [inline test/test_parser.py]: This check is failing and I'm not sure why.

Comment (ZeyadTamimi) [reply test/test_parser.py]: I have an idea, those two values are floats right? You need a fuzzy compare. Ill try to look at it today.

Comment (rouyeen) [reply test/test_parser.py]: maybe assertAlmostEqual() ?

**Comment by ZeyadTamimi** [comment]
test_db_add_correct_num_rows is still flaky

Also dont import import *

**Comment by ZeyadTamimi** [comment]
In the interest of getting people actual test data, I am merging this change :+1:

**Pull request merged**